src/notificationsReader.coffee

```diff
@@ -118,13 +121,18 @@ class NotificationsReader
         .catch (error) =>
           @_log "--> Error deleting message: #{error}. #{messageId}"
 
-  _notifyError: (message, error) => @_notify message, 'error', error
+  _notifyError: (message, error) => @_notify @statusObservers, message, 'error', error
+
+  _notifySuccess: (message) => @_notify @statusObservers, message, 'success'
+
+  _notifyFinish: (message) => @_notify @config.finishObservers, message, 'finish'
 
-  _notifySuccess: (message) => @_notify message, 'success'
+  _buildNotification: (message) =>
+    _.merge { message }, _.pick @config, ["app","topic","subscription"]
 
-  _notify: (message, event, opts) =>
-    notification = _.merge { message }, _.pick @config, ["app","topic","subscription"]
-    @observers.forEach (observer) => observer[event] notification, @, opts
+  _notify: (observers, message, event, opts) =>
+    notification = @_buildNotification message
+    observers?.forEach (observer) => observer[event] notification, @, opts
```

Comment:
Todos los observers manejan promises. Pero se puede agregar un `try` en la lambda del forEach si te parece. For if the flies.